backend/lib/endpoints/posts.js

```diff
@@ -370,10 +370,8 @@ async function routes(app) {
           "author.location.zip": false,
         }),
       );
-      if (postErr) {
-        req.log.error(postErr, "Failed retrieving post");
-        throw app.httpErrors.internalServerError();
-      } else if (post === null) {
+      if (post === null || postErr) {
```

Comment:
I think it's better to treat these 2 cases differently, please check the `user` [endpoint code as an example](https://github.com/FightPandemics/FightPandemics/blob/9eca8b54ad34bc433b51a4591123417fb5b23b19/backend/lib/endpoints/users.js#L217),

Reply:
<img width="694" alt="Screen Shot 2021-01-04 at 1 02 23 AM" src="https://user-images.githubusercontent.com/30244286/103513944-dc808280-4e28-11eb-9628-337e0669ba04.png">

Hi @ddveloper! Initially, as you can see above, the code was similar to the user endpoint code that you attached as an example. Resolving the given issue with this PR in two separate cases as it initially was would result in 2 cases that throw the same error. That's why I decided to merge them into one case since I didn't touch in the case when "post === null". But if you still think it's best to have them as two separate cases that throw the same error, I can go ahead and do that. Let me know

Reply:
If the ObjectId is invalid, exp "`GET /api/posts/1`"  this will result in `postErr` and not `post === null` which is why we get the 500 error.
But now we won't know when the db failed, since it will always return 404. A better thing to do will be validating the ObjectId and returning a 400 bad request error.

Reply:
Yeah now I get what you mean. It's just that 400 bad request didn't seem appropriate for the case where the ObjectId is invalid/doesn't exist and 404 seemed more appropriate. But you're right that having two cases that send the same response makes it hard to understand when the db failed.  I have changed the code as per your suggestion. Let me know if there's anything else to change. Thank you!